typescript/cli/src/rebalancer/config/Config.ts

```diff
@@ -0,0 +1,182 @@
+  withMetrics: z.boolean().optional(),
+  monitorOnly: z.boolean().optional(),
+  coingeckoApiKey: z.string().optional(),
+  rebalanceStrategy: z.enum(['weighted', 'minAmount']).optional(),
+});
+
+const ConfigSchema = BaseConfigSchema.catchall(ChainConfigSchema).superRefine(
+  (config, ctx) => {
+    // Get all chain names from the config
+    const chainNames = new Set(
+      Object.keys(config).filter(
+        (key) => !Object.keys(BaseConfigSchema.shape).includes(key),
+      ),
+    );
+
+    // Check each chain's overrides
+    for (const chainName of chainNames) {
+      const chain = config[chainName] as ChainConfig;
+
+      if (chain.override) {
+        for (const overrideChainName of Object.keys(chain.override)) {
+          // Each override key must reference a valid chain
+          if (!chainNames.has(overrideChainName)) {
+            ctx.addIssue({
+              code: z.ZodIssueCode.custom,
+              message: `Chain '${chainName}' has an override for '${overrideChainName}', but '${overrideChainName}' is not defined in the config`,
+              path: [chainName, 'override', overrideChainName],
+            });
+          }
+
+          // Override shouldn't be self-referencing
+          if (chainName === overrideChainName) {
+            ctx.addIssue({
+              code: z.ZodIssueCode.custom,
+              message: `Chain '${chainName}' has an override for '${chainName}', but '${chainName}' is self-referencing`,
+              path: [chainName, 'override', overrideChainName],
+            });
+          }
+        }
+      }
+    }
+  },
+);
+
+// Define separate types for each strategy config
+export type WeightedChainConfig = z.infer<typeof WeightedChainConfigSchema>;
+
+export type MinAmountChainConfig = z.infer<typeof MinAmountConfigSchema>;
+
+// Union type for all chain configs
+export type ChainConfig = z.infer<typeof ChainConfigSchema>;
+
+export type BaseConfig = z.infer<typeof BaseConfigSchema>;
+
+export class Config {
+  static load(
+    configFilePath: string,
+    rebalancerKey: string,
+    overrides: BaseConfig,
+  ) {
+    const config = readYamlOrJson(configFilePath);
+    const validationResult = ConfigSchema.safeParse(config);
+
+    if (!validationResult.success) {
+      throw new Error(fromZodError(validationResult.error).message);
+    }
+
+    const {
+      warpRouteId: fileWarpRouteId,
+      checkFrequency: fileCheckFrequency,
+      monitorOnly: fileMonitorOnly,
+      withMetrics: fileWithMetrics,
+      coingeckoApiKey: fileWithCoingeckoApiKey,
+      rebalanceStrategy: fileRebalanceStrategy,
+      ...chains
+    } = validationResult.data;
+
+    if (!Object.keys(chains).length) {
+      throw new Error('No chains configured');
+    }
+
+    const warpRouteId = overrides.warpRouteId ?? fileWarpRouteId;
+    const checkFrequency = overrides.checkFrequency ?? fileCheckFrequency;
+    const monitorOnly = overrides.monitorOnly ?? fileMonitorOnly ?? false;
```

Comment:
should these be defaulted in the schema?

Reply:
[`9fa5ebc` (#6321)](https://github.com/hyperlane-xyz/hyperlane-monorepo/pull/6321/commits/9fa5ebc8838bc52360fd2e6c7d72ffb52db765ea)